src/Backups/BackupEntryFromImmutableFile.cpp

```diff
@@ -59,16 +59,31 @@ UInt64 BackupEntryFromImmutableFile::getSize() const
 
 UInt128 BackupEntryFromImmutableFile::getChecksum() const
 {
-    std::lock_guard lock{size_and_checksum_mutex};
-    if (!checksum_adjusted)
     {
-        if (!checksum)
-            checksum = BackupEntryWithChecksumCalculation<IBackupEntry>::getChecksum();
```

Comment:
Here was a deadlock - `BackupEntryFromImmutableFile::getChecksum()` locked `size_and_checksum_mutex`, called `BackupEntryWithChecksumCalculation<IBackupEntry>::getChecksum()`, which called `BackupEntryFromImmutableFile::getSize()`, which tried to lock `size_and_checksum_mutex` again.